### DISCUSSION TITLE
Not responding after certain times.

### DISCUSSION THREAD

**Discussion by megabitsenmzq** [HomeSpan/HomeSpan#455] [Q&A] [answered]
I'm using a board called M5Atom to automate my AC. You can check the code here: https://github.com/megabitsenmzq/ESP32-HomeKit-Remote

I don't know if I did something wrong. But it always shows "Not responding" in my home app after certain times. When I check it from my wifi router, the router says it is connected and the signal is 100%. I have to add some code to reset the device every half hour to keep it working. But it still does not respond sometimes.

And I also find that if I keep the serial monitor in Arduino IDE open, this problem seems to disappear. That's kinda weird.

Here are other discussions like this: https://github.com/HomeSpan/HomeSpan/discussions/334 says it might be caused by naming conflicts, but I don't think I have any.

**Comment by DukemFIVE** [comment]
I have the same issues. I flash the SW, connect it to Homekit and after 2 min it shows no response.

I downgraded to version 1.5.1, now i dont have this issues anymore.

**Reply by HomeSpan** [reply]
Are you also using the M5Atom?  Does the problem resolve if you keep the Serial Monitor open?

**Reply by megabitsenmzq** [reply]
My board lasts longer, like 2~3 hours.

**Reply by megabitsenmzq** [reply]
Tried downgrade, not work for me.

**Comment by megabitsenmzq** [comment]
It seems I have just solved this issue by just moving the homeSpan.poll() from the top to the end of the loop function. 😂 I will continue to test it this week. I'm still curious why this can be fixed by just keeping the serial monitor open.

**Reply by megabitsenmzq** [reply]
The issue seems fixed by this. (Update: Nope😂)

**Reply by megabitsenmzq** [reply]
It stopped responding today when I try it. It seems if there are some delays between polls, HomeSpan will stop responding after certain times. Auto polls work. But it will interfere with the IR lib I'm using.

In my case, I have a delay(10);

**Reply by megabitsenmzq** [reply]
Update: Remove the delay did nothing. I even removed all M5 libraries, but still no luck.

**Comment by HomeSpan** [comment]
Does this issue occur for tutorial example 1?  Also, are you using HomeSpan 1.7.0 (latest version)?

**Reply by megabitsenmzq** [reply]
Yes, I'm on the latest version. Today I tried example 1 on another board. After several hours, the light bulb (example 1) and the AC (my code) stopped responding at the same time. And reboot any of them can bring them back to life (also at the same time)! I don't quite understand why 😂.

**Reply by megabitsenmzq** [reply]
And if I remove button related code (within the loop), my device can work for several hours before not responding. But if I bring these codes back, the device will stop responding in 30 minutes. (I'm not sure about this. Continue testing)

**Reply by HomeSpan** [reply]
Are you also using the latest version of the ESP32 Board Manager (should be version 2.0.5)?

**Reply by megabitsenmzq** [reply]
Yes, I'm on the latest version.

**Comment by megabitsenmzq** [comment]
After further investigation, I found that after the device stopped responding, the weblog became inaccessible ether. The URL (http://homespan-3274fd8eba8a.local/status) takes forever to load but using the IP address (http://10.0.0.18/status) I can still access the weblog. It seems it's a problem related to mDNS!

**Reply by megabitsenmzq** [reply]
And I can't really confirm any possible fixes I said before really works after many tests. (Using auto pull, keeping serial monitor open while using, etc...)

**Reply by megabitsenmzq** [reply]
Enabling IGMP Proxy in my router seems able to fix the problem. Continue testing.

**Comment by HomeSpan** [comment]
Do you have a mesh network?  At least one article I've read recommends enabling IGMP proxy to solve problems with HomeKit on mesh networks.

**Comment by megabitsenmzq** [comment, marked as answer]
I only have a single router, but enabling the IGMP proxy seems does fix the problems. The device usually stops responding within one hour, but now it already lasts a whole day. I guess this is the solution!

**Reply by megabitsenmzq** [reply]
After another whole day, it still works. Fixed✌️
